components/org.wso2.carbon.identity.api.server.authenticators/org.wso2.carbon.identity.api.server.authenticators.v1/src/main/java/org/wso2/carbon/identity/api/server/authenticators/v1/impl/LocalAuthenticatorConfigBuilderFactory.java

```diff
@@ -147,4 +148,16 @@ private static AuthenticatorPropertyConstants.AuthenticationType resolveAuthenti
             return AuthenticatorPropertyConstants.AuthenticationType.IDENTIFICATION;
         }
     }
+
+    private static void validateUserDefinedLocalAuthenticatorConfig(UserDefinedLocalAuthenticatorCreation config)
+            throws AuthenticatorMgtClientException {
+
+        if (config.getEndpoint().getAuthentication().getProperties() == null ||
```

Comment:
what if authentication type is null ?
Can't we move this to buildEndpointConfig() method because that's where authentication should be validated

Reply:
if the authentication type is null, that will be validated in here.
https://github.com/wso2/carbon-identity-framework/blob/cc93d8400bc95e142af2b8d2ac2dc693b4be0a6a/components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/UserDefinedAuthenticatorEndpointConfig.java#L141

Reply:
If we are to move this `buildEndpointConfig()` method we have to break that method into two, as this validation should not be applied in the update flow

Reply:
ack. let's proceed